apps/aecore/lib/aecore/contract/vm_chain.ex

```diff
@@ -0,0 +1,134 @@
+defmodule Aecore.Contract.VmChain do
+  @moduledoc """
+  VM chain api implementation for interaction with our chain
+  """
+
+  alias Aecore.Contract.ContractStateTree
+  alias Aecore.Contract.CallStateTree
+  alias Aecore.Contract.Contract
+  alias Aecore.Contract.Call
+  alias Aecore.Chain.Chainstate
+  alias Aecore.Chain.Worker, as: Chain
+  alias Aecore.Account.Account
+  alias Aecore.Tx.DataTx
+  alias Aecore.Keys
+  alias Aevm.ChainApi
+
+  @behaviour Aevm.ChainApi
+
+  @spec new_state(Keys.pubkey(), Chainstate.t()) :: ChainApi.chain_state()
+  def new_state(pubkey, chain_state) do
+    %{pubkey: pubkey, chain_state: chain_state}
+  end
+
+  @spec get_balance(Keys.pubkey(), ChainApi.chain_state()) :: non_neg_integer()
+  def get_balance(pubkey, state) do
+    account_tree = state.accounts
+    Account.balance(account_tree, pubkey)
+  end
+
+  @spec get_store(ChainApi.chain_state()) :: ChainApi.store()
+  def get_store(%{pubkey: pubkey, chain_state: chain_state}) do
+    contract_tree = chain_state.contracts
+
+    case ContractStateTree.get_contract(contract_tree, pubkey) do
+      %Contract{} = contract ->
+        store = Contract.store(contract)
+
+        Enum.reduce(store, %{}, fn {key, value}, acc ->
```

Comment:
Isn't this the same as `Aevm.State.storage_to_int` function? I think the deserialization of int-s should be part of `ContractStateTree.get_contract` probably.